### DISCUSSION TITLE
Ram and Vram usage so hight

### DISCUSSION THREAD

**Discussion by shpinog** [open-webui/open-webui#7791] [Issues]
Hello, I'm still a beginner when it comes to everything related to LLVM. My hardware is RX6800 + R7-3700X and 32GB RAM.

I use open-webui + ollama and encounter some strange behavior with models like 4Q and others around 12-13GB in size. Some models in the chat behave quite oddly, they may respond to one request or a few normally, after which they go crazy and might output "@#&&#*" or something similar.

At the same time, I notice that the 22B models take up quite a lot of VRAM, yet there is still around 1-2GB of free space. However, smaller models also take up quite a bit. The context size is set between 2048-4096 (varies).

I thought this was normal since other applications using ollama behave in a similar way, but I tried running these models with coboldcpp, using Vulkan and 40 GPU layers, and I get VRAM consumption of about 10 000MB at startup and a small RAM usage of about 5000MB (virt) and using context  8196!!!  The model runs quite fast, and visually there’s no significant difference, everything happens interactively in the chat. At the same time, there are no problems with work and any “@#&&#*”.

My question is, how can I achieve this in openwebui? I understand that openwebui is essentially just a frontend over ollama in this case, but is there a way to use a more efficient and resource-optimized backend? Thank you.

**Comment by mahenning** [comment]
I can't speak about kobold.cpp, but I use vLLM as an endpoint, which uses the OpenAI API. I assume kobold cpp can be also used that way.  
You can go to Admin settings -> Connections and enter `http://[ip-where-endpoint-runs]:8080/v1` with an empty API Key field. If you run the endpoint on your local machine, maybe `localhost` works as an ip replacement, and you also have to look up if kobold cpp uses another port than 8080 (I think I saw 5001 in their docs)
